deposit/src/main/java/edu/unc/lib/deposit/fcrepo4/IngestContentObjectsJob.java

```diff
@@ -0,0 +1,449 @@
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.deposit.fcrepo4;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.hp.hpl.jena.rdf.model.Bag;
+import com.hp.hpl.jena.rdf.model.Model;
+import com.hp.hpl.jena.rdf.model.ModelFactory;
+import com.hp.hpl.jena.rdf.model.NodeIterator;
+import com.hp.hpl.jena.rdf.model.Property;
+import com.hp.hpl.jena.rdf.model.Resource;
+import com.hp.hpl.jena.rdf.model.Statement;
+import com.hp.hpl.jena.vocabulary.DC;
+import com.hp.hpl.jena.vocabulary.RDF;
+
+import edu.unc.lib.deposit.work.AbstractDepositJob;
+import edu.unc.lib.deposit.work.DepositGraphUtils;
+import edu.unc.lib.dl.fcrepo4.ContentObject;
+import edu.unc.lib.dl.fcrepo4.FileObject;
+import edu.unc.lib.dl.fcrepo4.FolderObject;
+import edu.unc.lib.dl.fcrepo4.PIDs;
+import edu.unc.lib.dl.fcrepo4.WorkObject;
+import edu.unc.lib.dl.fedora.FedoraException;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.rdf.Cdr;
+import edu.unc.lib.dl.rdf.CdrDeposit;
+import edu.unc.lib.dl.reporting.ActivityMetricsClient;
+import edu.unc.lib.dl.util.DepositException;
+import edu.unc.lib.dl.util.RedisWorkerConstants.DepositField;
+
+/**
+ * Ingests all content objects in the deposit into the Fedora repository.
+ * 
+ * @author bbpennel
+ *
+ */
+public class IngestContentObjectsJob extends AbstractDepositJob {
+	private static final Logger log = LoggerFactory.getLogger(IngestContentObjectsJob.class);
+
+	private boolean resumed;
+
+	// The set of object paths from this deposit which already exist in the repository
+	private Set<String> previouslyIngestedSet;
+
+	@Autowired
+	private ActivityMetricsClient metricsClient;
+
+	public IngestContentObjectsJob() {
+		super();
+	}
+
+	public IngestContentObjectsJob(String uuid, String depositUUID) {
+		super(uuid, depositUUID);
+	}
+
+	/**
+	 * Determines how many operations will be required to complete this deposit.
+	 * 
+	 * @param depositBag
+	 */
+	private void calculateWorkRemaining(Bag depositBag) {
+		List<String> contentPaths = new ArrayList<>();
+		DepositGraphUtils.walkChildrenDepthFirst(depositBag, contentPaths, true);
+		// Number of actions is the number of ingest objects plus deposit record
+		setTotalClicks(contentPaths.size());
```

Comment:
Currently the deposit record is added via a separate job, although i had been considering merging them together.  Either way though, I've found the inclusion of the deposit record in the object count to be rather confusing, particularly when depositing a single file, so I was leaning towards not reporting it that way.